implementations/rust/ockam/ockam_node/src/processor_builder.rs

```diff
@@ -4,98 +4,172 @@ use crate::{relay::ProcessorRelay, Context, NodeMessage};
-    ) -> Self {
-        let mailboxes = Mailboxes::main(
-            address.into(),
-            incoming_access_control,
-            outgoing_access_control,
-        );
-
-        Self {
-            mailboxes,
-            processor,
+    /// Worker with only one [`Address`]
+    pub fn with_address(self, address: impl Into<Address>) -> ProcessorBuilderOneAddress<P> {
+        ProcessorBuilderOneAddress {
+            incoming_ac: Arc::new(DenyAll),
+            outgoing_ac: Arc::new(DenyAll),
+            processor: self.processor,
+            address: address.into(),
         }
     }
 
-    /// Create a processor which uses the access control from the given
-    /// [`Mailboxes`]
-    pub fn with_mailboxes(mailboxes: Mailboxes, processor: P) -> Self {
-        Self {
+    /// Worker with multiple [`Address`]es
+    pub fn with_mailboxes(self, mailboxes: Mailboxes) -> ProcessorBuilderMultipleAddresses<P> {
+        ProcessorBuilderMultipleAddresses {
             mailboxes,
-            processor,
+            processor: self.processor,
         }
     }
+}
 
+pub struct ProcessorBuilderMultipleAddresses<P>
+where
+    P: Processor<Context = Context>,
+{
+    mailboxes: Mailboxes,
+    processor: P,
+}
+
+impl<P> ProcessorBuilderMultipleAddresses<P>
+where
+    P: Processor<Context = Context>,
+{
     /// Consume this builder and start a new Ockam [`Processor`] from the given context
-    #[inline]
-    pub async fn start(self, context: &Context) -> Result<Address> {
-        info!(
-            "Initializing ockam processor '{}' with access control in:{:?} out:{:?}",
-            self.mailboxes.main_address(),
-            self.mailboxes.main_mailbox().incoming_access_control(),
-            self.mailboxes.main_mailbox().outgoing_access_control(),
-        );
-
-        let mailboxes = self.mailboxes;
-        let main_address = mailboxes.main_address().clone();
-
-        // Pass it to the context
-        let (ctx, sender, ctrl_rx) = context.copy_with_mailboxes(mailboxes);
-
-        debugger::log_inherit_context("PROCESSOR", context, &ctx);
-
-        // Then initialise the processor message relay
-        ProcessorRelay::<P>::init(context.runtime(), self.processor, ctx, ctrl_rx);
-
-        // Send start request to router
-        let (msg, mut rx) = NodeMessage::start_processor(main_address.clone(), sender);
-        context
-            .sender()
-            .send(msg)
-            .await
-            .map_err(|e| Error::new(Origin::Node, Kind::Invalid, e))?;
-
-        // Wait for the actual return code
-        rx.recv()
-            .await
-            .ok_or_else(|| NodeError::NodeState(NodeReason::Unknown).internal())??;
-
-        Ok(main_address)
+    pub async fn start(self, context: &Context) -> Result<()> {
+        start(context, self.mailboxes, self.processor).await
     }
 }
+
+pub struct ProcessorBuilderOneAddress<P>
+where
+    P: Processor<Context = Context>,
+{
+    incoming_ac: Arc<dyn IncomingAccessControl>,
```

Comment:
I wonder if we can just default to DenyAll for incoming access control in processors